### PR TITLE
Revise deprecations

### PR DESCRIPTION
**Pull request by bdice**
## Description
Revises deprecations discussed with @vyasr via Slack. We want to move deprecations targeting 1.0 to future 0.x versions, because our roadmap to 1.0 is not yet clear. Of course, if 1.0 comes early, we're still technically correct to say that it will be removed in a 0.x version (because 1.0 > 0.x).

I also deprecated the function `make_bundles` and replaced it with a private method.

## Motivation and Context
Clarifying software roadmap.

## Types of Changes

- [ ] Documentation update
- [ ] Bug fix
- [x] New feature? Not sure.
- [ ] Breaking change<sup>1</sup>

<sup>1</sup>The change breaks (or has the potential to break) existing functionality.

## Checklist:

- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/signac-flow/blob/master/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/signac-flow/blob/master/contributors.yaml).
- [x] My code follows the [code style guideline](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md#code-style) of this project.
- [x] The changes introduced by this pull request are covered by existing or newly introduced tests.

If necessary:
- [ ] I have updated the API documentation as part of the package doc-strings.
- [ ] I have created a separate pull request to update the [framework documentation](https://docs.signac.io/) on [signac-docs](https://github.com/glotzerlab/signac-docs) and linked it here.
- [ ] I have updated the [changelog](https://github.com/glotzerlab/signac-flow/blob/master/changelog.txt).

### CODE REVIEW

File: flow/project.py [hunk 1]
```diff
@@ -128,7 +128,7 @@ def false(cls, key):
 
     @classmethod
     @deprecated(
-        deprecated_in="0.9", removed_in="1.0",
+        deprecated_in="0.9", removed_in="0.11",
```
Comment (zhou-pj) [inline flow/project.py]: Should we add `, current_version=__version__` for consistency?

Comment (zhou-pj) [inline flow/project.py]: Just curious, is the reason for removing in 0.11 here vs 0.10 as most others being that we usually make deprecated and removed version to be 2 versions apart?

Comment (bdice) [reply flow/project.py]: Good find! I hadn't noticed that `current_version` was missing. I fixed it.

Comment (bdice) [reply flow/project.py]: Yes, that is why. We have a deprecation cycle of 2 minor versions: https://docs.signac.io/projects/flow/en/latest/support.html#deprecation-policy

**Pull request merged**